pkg/bus/bus.go

```diff
@@ -205,6 +206,16 @@ func (b *InProcBus) AddEventListener(handler HandlerFunc) {
 	b.listeners[eventName] = append(b.listeners[eventName], handler)
 }
 
+func (b *InProcBus) AddEventListenerCtx(handler HandlerFunc) {
```

Comment:
addeventlistenerCtx is exactly the same as AddEventlistener, why not using just AddEventListener in this case?

Reply:
in one of the handlers I added (service.go) I propagated context as a first parameter of function signature, that's why added AddEventListenerCtx, on 211(bus.go) it reads the second parameter as handler type and not the first. Position 1 instead of 0.

Reply:
Then you probably need to add a PublishCtx as well. Maybe do these changes in a separate PR?

Reply:
I need to add it for DispatchCtx and I'm trying to reduce context.TODO()

Reply:
Can't you just skip changing use of `AddEventListener`?

Reply:
ok, I can split this into to PRs